model-validation-track/ModelValidator.py

```diff
@@ -5,41 +5,24 @@
 from pysmt.shortcuts import get_env
 from pysmt.smtlib.parser import SmtLibParser, Tokenizer
 from pysmt.exceptions import PysmtSyntaxError
+from pysmt.smtlib.utils import SmtLibModelValidationSimplifier
 
-
-def parseDefineFun(parser, tokens):
-    current = tokens.consume()
-
-    if (current != "define-fun"):
-        raise PysmtSyntaxError("'define-fun' expected", tokens.pos_info)
-    var = parser.get_expression(tokens)
-    if (not var.is_symbol()):
-        raise PysmtSyntaxError("Symbol expected", tokens.pos_info)
-    namedparams = parser.parse_named_params(tokens, "define-fun")
-    if (namedparams):
-        raise PysmtSyntaxError("'()' expected", tokens.pos_info)
-    rtype = parser.parse_type(tokens, "define-fun")
-    ebody = parser.get_expression(tokens)
-    if (not ebody.is_constant()):
-        raise PysmtSyntaxError("Constant expected", tokens.pos_info)
-    current = tokens.consume()
-    return (var, ebody)
-
+get_env().allow_empty_var_names = True
 
 def readModel(parser, modelFile, inputFile):
     with open(modelFile) as script:
-        model = {}
-        symbols = parser.env.formula_manager.symbols
-        parser.cache.update(symbols)
-        tokens = Tokenizer(script, interactive=parser.interactive)
-        res = []
-        current = tokens.consume()
-
-        if (current == "unknown"):
+        lino = 0
+        for line in script:
+            lino += 1
+            read_status = line.strip()
+            if read_status != 'success':
```

Comment:
When the pre-processor is changed for SMT-COMP 2021, there should be no `success` in the output.
But we can leave that in for now.

Reply:
OK, I added this just because of this example model: https://github.com/SMT-COMP/postprocessors/blob/master/model-validation-track/examples/QF_UF/model1-smtinterpol.smt2

We can just remove the for and use a `getline()` instead to read the `sat/unsat/unknown`